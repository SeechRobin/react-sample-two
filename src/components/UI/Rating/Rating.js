import React, { Component } from "react";
import PropTypes from "prop-types";

class Rating extends Component {
  render() {
    const { rating } = this.props;

    const stars = Math.round(rating);
    let reviewsStars = [];
    for (let i = 1; i <= 5; i++) {
      i <= stars
        ? reviewsStars.push(<span key={i} className="fa fa-star checked" />)
        : reviewsStars.push(<span key={i} className="fa fa-star unchecked" />);
    }

    return <div>{reviewsStars}</div>;
  }
}

export default Rating;

Rating.propTypes = {
  rating: PropTypes.number
};
