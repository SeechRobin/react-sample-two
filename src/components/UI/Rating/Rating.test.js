import React from "react";
import { shallow } from "enzyme";

import Rating from "./Rating";

describe("<Rating />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Rating rating={3} />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
