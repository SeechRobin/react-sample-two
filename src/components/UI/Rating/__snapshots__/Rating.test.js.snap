// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rating
    rating={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="fa fa-star checked"
        />,
        <span
          className="fa fa-star checked"
        />,
        <span
          className="fa fa-star checked"
        />,
        <span
          className="fa fa-star unchecked"
        />,
        <span
          className="fa fa-star unchecked"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-star checked",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-star checked",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-star checked",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-star unchecked",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-star unchecked",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="fa fa-star checked"
          />,
          <span
            className="fa fa-star checked"
          />,
          <span
            className="fa fa-star checked"
          />,
          <span
            className="fa fa-star unchecked"
          />,
          <span
            className="fa fa-star unchecked"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-star checked",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-star checked",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-star checked",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-star unchecked",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-star unchecked",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
